tests/test.cpp

```diff
@@ -0,0 +1,104 @@
+#include <blackbox.h>
+#include <gtest/gtest.h>
+
+#include <atomic>
+#include <csignal>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+namespace {
+
+std::atomic_bool signal_nesting_error;
+std::atomic_bool signal_thread_started;
+
+void SignalHandler(int signal) {
+  if (signal == SIGUSR1) {
+    if (blackbox::write("test") == -EDEADLK) {
+      signal_nesting_error.store(true);
```

Comment:
Just for intellectual fun, this could be std::memory_order_relaxed, right? Same below